paddle/fluid/imperative/layer.cc

```diff
@@ -316,19 +328,14 @@ void PyLayer::RegisterFunc(int func_id, const py::object& py_func) {
 
 int PyLayer::NumFuncs() { return py_funcs_.size(); }
 
-std::vector<VarBase*> PyLayer::Apply(int func_id,
-                                     const std::vector<VarBase*>& inputs) {
+std::vector<Variable*> PyLayer::Apply(int func_id,
```

Comment:
why change this? pylayer can be followed by other layers?

Reply:
Because every VarBase should have a name now.. so VarBase's construnction was moved into PyTrace